### PR TITLE
Fix tests remote debugging and improved instructions

### PR DESCRIPTION
**Pull request by jkasten2**
# Description
Fix tests remote debugging and improved instructions.

## Details
See [render tests/README.md](https://github.com/OneSignal/OneSignal-Website-SDK/blob/tests/allow_debugging/test/README.md)

# Systems Affected
* [x] Dev Env of WebSDK

# Validation
### Checklist
* [x] All the automated tests pass or I explained why that is not possible
* [x] I have personally tested this on my machine or explained why that is not possible
* [x] I have included test coverage for these changes or explained why they are not needed

**Programming Checklist**
Typescript:

* [x] No Typescript warnings
* [x] Avoid silencing null/undefined warnings with the exclamation point

### Info
image

### Checklist
* [x] I have included screenshots/recordings of the intended results or explained why they are not needed

- - -
## Related Tickets
- - -

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-website-sdk/766)

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -1,72 +1,38 @@
...
-## Testing Commands
-1. Run all and watch file changes: `yarn test`
-2. Run all with no watch: `yarn test:noWatch`
-3. Run specific file: `<one of above commands> <path to transpiled file>` e.g: `yarn test:noWatch build/ts-to-es6/test/unit/managers/ConfigManager.js`
+## VSCode Run a specific test
+_Tested up to VSCode Version: `1.54.2`_
+1. Install the ["Remote Development"](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) VS Code Extension
+   - Restart VSCode
+2. You will be prompted that there is a Dev Container detected, press "reopen in a container"
+   - If no prompt, press F1 and enter "Remote-Containers: Open Folder in Container.."
```
Comment (rgomezp) [inline test/README.md:8]: Provide Mac shortcut as well or say "Open the Command Palette"

File: test/README.md [hunk 2]
```diff
@@ -1,72 +1,38 @@
...
+3. Install [Testify](https://marketplace.visualstudio.com/items?itemName=felixjb.testify) VS Code Extension
+4. Go to the test file you want to run and click "Run Test" above the name.
```
Comment (rgomezp) [inline test/README.md:10]: I'm not seeing this above the file name. Perhaps provide a screenshot to be clearer.

File: test/README.md [hunk 3]
```diff
@@ -1,72 +1,38 @@
-# Testing the WebSDK
+# Running WebSDK Test Suite
 
-## Testing Commands
-1. Run all and watch file changes: `yarn test`
-2. Run all with no watch: `yarn test:noWatch`
-3. Run specific file: `<one of above commands> <path to transpiled file>` e.g: `yarn test:noWatch build/ts-to-es6/test/unit/managers/ConfigManager.js`
+## VSCode Run a specific test
+_Tested up to VSCode Version: `1.54.2`_
+1. Install the ["Remote Development"](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) VS Code Extension
```
Comment (rgomezp) [inline test/README.md:5]: More context might be useful here (think from the perspective of a newhire). Why are we running in the docker container? e.g: does this relate to transpilation or build process?

File: test/README.md [hunk 4]
```diff
@@ -1,72 +1,38 @@
...
+
+If you need to configure any advanced parameters in the test edit `.vscode/launch.json`
+
+## Testing from CLI
+1. Run all and watch file changes: `yarn test`
+2. Run all with no watch: `yarn test:noWatch`
+3. Run specific file: `<one of above commands> <path to file>`
+   - e.g: `yarn test:noWatch test/unit/context/sw/ServiceWorker.ts`
+3. Run specific test in a file: `<one of above commands> <path to file> -m "name of test here"`
+   - e.g: `yarn test:noWatch test/unit/context/sw/ServiceWorker.ts -m "onPushReceived - Ensure undefined payload does not show"`
```
Comment (rgomezp) [inline test/README.md:38]: I got the following when running the test:
```
Debugger attached.

  ✖ Provide the path to the test file you wish to debug
Waiting for the debugger to disconnect...
```

Let's add a troubleshooting section for the solution to this as others may get the same in the future

File: test/README.md [hunk 5]
```diff
@@ -1,72 +1,38 @@
...
-            "outputCapture": "std",
-            "port": 9229,
-            "sourceMaps": true,
-            "trace": true,
-            "protocol": "auto",
-            "console": "integratedTerminal"
-        },
-    ]
-}
-```
```
Comment (rgomezp) [inline test/README.md:70]: Why is this removed? The default way to run tests is using test launch files.

Perhaps add a general overview section where we explain that we are using a different environment via these plugins.

File: test/README.md [hunk 6]
```diff
@@ -1,72 +1,38 @@
-# Testing the WebSDK
+# Running WebSDK Test Suite
 
-## Testing Commands
-1. Run all and watch file changes: `yarn test`
-2. Run all with no watch: `yarn test:noWatch`
-3. Run specific file: `<one of above commands> <path to transpiled file>` e.g: `yarn test:noWatch build/ts-to-es6/test/unit/managers/ConfigManager.js`
+## VSCode Run a specific test
+_Tested up to VSCode Version: `1.54.2`_
```
Comment (rgomezp) [inline test/README.md:4]: Provide plugin versions as well

**Review by rgomezp — COMMENTED** [review]
I was unable to get this running.

There's also some more context that can be added

File: test/README.md (hunk 4 above)
Comment (jkasten2) [reply test/README.md:50]: > I got the following when running the test:

I didn't have any issue with a debugger not attaching when running a test from the terminal inside of VScode. However after testing I noticed it takes about 2x longer to start so I would recommend debugging from VSCode noted above.

> Let's add a troubleshooting section for the solution to this as others may get the same in the future

Not sure what to add here, not sure why that didn't work for but since it isn't the recommend way to debug I don't see it as a blocker.

**Comment by rgomezp** [comment]
Just putting a reminder here to add a (troubleshooting?) section talking about the weird docker on Mac issue I was seeing

**Review by rgomezp — APPROVED** [review]
